frontend/src/app/modules/external-booking/external-booking-choose-time-page/external-booking-choose-time-page.component.ts

```diff
@@ -82,64 +114,104 @@ export class ExternalBookingTimeComponent extends BaseComponent implements OnIni
+    public AddTimeAndDate(timeIndex: number, dayIndex: number, timeZone: TimeZone): void {
+        const date = addDays(this.calendarWeek.firstDay, dayIndex);
+        const time = addHours(this.theEarliestStartOfTimeRanges, (this.selectedMeetingDuration * timeIndex) / 60);
 
-        this.selectedTimeAndDateEvent.emit({ date, timeFinish });
+        date.setHours(time.getHours());
+        date.setMinutes(time.getMinutes());
+        date.setSeconds(time.getSeconds());
+        date.setMilliseconds(time.getMilliseconds());
+        const timeFinish = new Date(time.getTime() + this.selectedMeetingDuration * 60000);
+
+        this.selectedTimeAndDateEvent.emit({ date, timeFinish, timeZone });
     }
 
     public changeWeek(addingMode: boolean): void {
         if (addingMode) {
-            this.calendarWeek.firstDay = new Date(
-                this.calendarWeek.firstDay.setDate(this.calendarWeek.firstDay.getDate() + 7),
-            );
-            this.calendarWeek.lastDay = new Date(
-                this.calendarWeek.lastDay.setDate(this.calendarWeek.lastDay.getDate() + 7),
-            );
+            this.calendarWeek.firstDay = addDays(this.calendarWeek.firstDay, 7);
+            this.calendarWeek.lastDay = addDays(this.calendarWeek.lastDay, 7);
         } else {
-            this.calendarWeek.firstDay = new Date(
-                this.calendarWeek.firstDay.setDate(this.calendarWeek.firstDay.getDate() - 7),
-            );
-            this.calendarWeek.lastDay = new Date(
-                this.calendarWeek.lastDay.setDate(this.calendarWeek.lastDay.getDate() - 7),
-            );
+            this.calendarWeek.firstDay = subDays(this.calendarWeek.firstDay, 7);
+            this.calendarWeek.lastDay = subDays(this.calendarWeek.lastDay, 7);
         }
     }
 
     public isTodayDate(date: Date, daysToAdd: number = 0): boolean {
+        const newDate = addDays(date, daysToAdd);
+
         return (
-            date.getDate() + daysToAdd === this.nowDate.getDate() &&
-            date.getMonth() === this.nowDate.getMonth() &&
-            date.getFullYear() === this.nowDate.getFullYear()
+            newDate.getDate() === this.nowDate.getDate() &&
+            newDate.getMonth() === this.nowDate.getMonth() &&
+            newDate.getFullYear() === this.nowDate.getFullYear()
         );
     }
 
-    public isDateInRange(date: Date, min: Date, max: Date, daysToAdd: number = 0): boolean {
+    public isDateInRange(date: Date, min: Date, max: Date, daysToAdd: number = 0, timesToAdd = 0): boolean {
+        let isOrdered = false;
+
+        const firstCalendarDay = new Date(this.calendarWeek.firstDay);
+
+        firstCalendarDay.setHours(date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds());
+        const result = addDays(firstCalendarDay, daysToAdd);
+
+        result.setTime(result.getTime() + this.selectedMeetingDuration * timesToAdd * 60 * 1000);
+
+        this.orderedTimes.forEach((el) => {
```

Comment:
Cannot findIndex (or some, if you wish) be used here?

Reply:
We cannot use it ( We must check every element in this.orderedTimes because it could be any date in any order.